client/components/community-translator/translatable.jsx

```diff
@@ -0,0 +1,214 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { isEmpty } from 'lodash';
+import { translate } from 'i18n-calypso';
+import classNames from 'classnames';
+import Gridicon from 'gridicons';
+
+/**
+ * Internal dependencies
+ */
+import Dialog from 'components/dialog';
+import Button from 'components/button';
+import TranslatableField from './translatable-field';
+import TranslatedSuccess from './translated-success';
+import { getTranslationData, getTranslationPermaLink, submitTranslation } from './utils.js';
+
+export class Translatable extends Component {
+	state = {
+		showDialog: false,
+		originalData: {},
+		formState: {},
+	};
+
+	hasDataLoaded() {
+		return ! isEmpty( this.state.originalData );
+	}
+
+	handleTranslationChange = event => {
+		const { name, value } = event.target;
+		this.setState( {
+			formState: {
+				...this.state.formState,
+				[ name ]: value,
+			},
+			submitting: false,
+			submissionSuccess: false,
+		} );
+	};
+
+	closeDialog = () => {
+		this.setState( {
+			showDialog: false,
+			submissionSuccess: false,
+			formState: this.state.submissionSuccess ? {} : this.state.formState,
+			originalData: this.state.submissionSuccess ? {} : this.state.originalData,
+		} );
+	};
+
+	openDialog = event => {
+		event.preventDefault();
+
+		this.setState( { showDialog: true } );
+
+		const { singular, context, plural, locale } = this.props;
+		! this.hasDataLoaded() &&
+			getTranslationData( locale, { singular, context, plural } ).then( originalData =>
+				this.setState( {
+					error: originalData.error,
+					originalData,
+					translationUrl: getTranslationPermaLink( originalData.originalId, locale ),
+					formState: {
+						translatedSingular: originalData.translatedSingular,
+						translatedPlural: originalData.translatedPlural,
+					},
+				} )
+			);
+	};
+
+	submitForm = () => {
+		this.setState( {
+			submitting: true,
+		} );
+		submitTranslation(
+			this.state.originalData.originalId,
+			this.state.formState,
+			this.props.locale
+		).then( originalData => {
+			this.setState( {
+				error: originalData.error,
+				originalData,
+				submitting: false,
+				submissionSuccess: true,
+			} );
+		} );
+	};
+
+	getDialogButtons = () => {
+		const buttons = [
```

Comment:
I'm not quite sure, but I think a dialog should only have one primary button in general. Having two would look like this:
<img width="536" alt="2018-04-11 4 29 08" src="https://user-images.githubusercontent.com/1842898/38608935-58010bba-3daf-11e8-936d-6046269f8e19.png">
which I think is a bit confusing and not good for accessibility.

Reply:
In that case, should we have some explicit way of telling the user how one can close the dialogue/cancel?

Hitting escape works. Or a close `x` ?

Reply:
A close `x` sounds good. One alternative is that we keep the close button, but remove its primary attribute when the submit button is enabled. 

I don't have a strong preference here, so I'll trust your best judgement here as long as there is no multiple primary buttons :)